File: gitlfs-client/src/main/java/ru/bozaro/gitlfs/client/io/StreamProvider.java
package ru.bozaro.gitlfs.client.io;

import javax.annotation.Nonnull;
import java.io.IOException;
import java.io.InputStream;

/**
 * Stream provider.
 *
 * @author Artem V. Navrotskiy
 */
@FunctionalInterface
public interface StreamProvider {
  @Nonnull
  InputStream getStream() throws IOException;
}

File: gitlfs-client/src/main/kotlin/ru/bozaro/gitlfs/client/io/StreamProvider.kt
package ru.bozaro.gitlfs.client.io

import java.io.IOException
import java.io.InputStream

/**
 * Stream provider.
 *
 * @author Artem V. Navrotskiy
 */
interface StreamProvider {
    @get:Throws(IOException::class)
    val stream: InputStream
}
